tests/versioning_3_test.go

```diff
@@ -232,6 +233,90 @@ func (s *Versioning3Suite) TestQueryWithPinnedOverride_Sticky() {
 	)
 }
 
+func (s *Versioning3Suite) TestPinnedQuery_DrainedVersion_PollersAbsent() {
+	s.RunTestWithMatchingBehavior(
+		func() {
+			s.testPinnedQuery_DrainedVersion(false)
+		},
+	)
+}
+
+func (s *Versioning3Suite) TestPinnedQuery_DrainedVersion_PollersPresent() {
```

Comment:
can you add a test for the scenario where a version becomes drained, and then becomes activated again (current or ramping), and then the query succeeds?